snapshotter/app/service.go

```diff
@@ -54,12 +54,19 @@ func Run(config config.Config) error {
 
 	group, ctx := errgroup.WithContext(ctx)
 
-	cache := cache.NewSnapshotterCache()
-
 	var (
 		monitor          *metrics.Monitor
 		serviceDiscovery *discovery.ServiceDiscovery
 	)
+
+	cache, err := initCache(config, monitor)
+	if err != nil {
+		log.G(ctx).WithFields(
+			logrus.Fields{"resolver": config.Snapshotter.Proxy.Address.Resolver.Type},
+		).WithError(err).Fatal("failed creating socket resolver")
+		return err
```

Comment:
Does it make sense to include this information into the `err` and pass that to the caller?